Podverse/PlaylistDetailTableViewController.swift

```diff
@@ -0,0 +1,153 @@
+                self.reloadPlaylistData(playlist: playlist)
+            }
+        }
+
+    }
+    
+    func reloadPlaylistData(playlist: Playlist?) {
+        if self.reachability.hasInternetConnection() == false {
+            self.showStatusMessage(message: "You must connect to the internet to load this playlist.")
+            return
+        }
+        
+        if let playlist = playlist {
+            self.itemCount.text = "Items: ###"
+            self.lastUpdated.text = playlist.lastUpdated?.toShortFormatString()
+            self.playlistTitle.text = playlist.title
+            self.mediaRefsArray = playlist.mediaRefs
+        }
+        
+        self.showPlaylistView()
+        self.tableView.reloadData()
+        
+    }
+    
+    func showStatusMessage(message: String) {
+        self.activityIndicator.stopAnimating()
+        self.statusMessage.text = message
+        self.tableView.isHidden = true
+        self.loadingView.isHidden = false
+        self.statusMessage.isHidden = false
+
+        if message == "You must connect to the internet to load this playlist." {
+            self.retryButton.isHidden = false
+        }
+    }
+    
+    func showIndicator() {
+        self.activityIndicator.startAnimating()
+        self.tableView.isHidden = true
+        self.loadingView.isHidden = false
+        self.activityIndicator.isHidden = false
+        self.statusMessage.isHidden = true
+        self.retryButton.isHidden = true
+    }
+    
+    func showPlaylistView() {
+        self.activityIndicator.stopAnimating()
+        self.tableView.isHidden = false
+        self.loadingView.isHidden = true
+        self.statusMessage.isHidden = true
+        self.retryButton.isHidden = true
+    }
+
+}
+
+extension PlaylistDetailTableViewController:UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.mediaRefsArray.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let mediaRef = self.mediaRefsArray[indexPath.row]
+        let cell = self.tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! PlaylistDetailTableViewCell
+
+        cell.episodeTitle.text = mediaRef.episodeTitle
+        cell.podcastTitle.text = mediaRef.podcastTitle
+        cell.pubDate.text = mediaRef.episodePubDate?.toShortFormatString()
+        
+        Podcast.retrievePodcastImage(podcastImageURLString: mediaRef.podcastImageUrl) { (podcastImage) -> Void in
+            DispatchQueue.main.async {
```

Comment:
This is probably not needed any more since retrieve dispatches to main queue now anyway